packages/core/admin/admin/src/pages/AuthPage/index.js

```diff
@@ -175,8 +154,8 @@ const AuthPage = ({ hasAdmin, setHasAdmin }) => {
       push('/');
     } catch (err) {
       if (err.response) {
+        setSubmitting(false);
```

Comment:
I'm not 100% convinced this is the right place to but the state transition 😬 . To me, it almost looks like this function could be used outside react without too much effort. Not sure it should own react logic, WYT?

Reply:
These functions `setSubmitting & setErrors` are provided by Formik. It is used to define the Formik context variables `isSubmitting` and `isValidating` used in our custom `Form`. Does it answer the question? 😅

Reply:
Sorry, what I meant here is that the function where this hook is used could be extracted and used in any kind of project with the same behavior. What I mean, is that calling `login()` is not bound to React itself so why would we bind React to it?

As a suggestion, what would you think of, depending on the value returned by the function, only then in the react component, you call the hooks functions?